pheweb/serve/data_access/db.py

```diff
@@ -638,42 +870,116 @@ def __init__(self, matrix_path):
                                     "stop_lost",
                                     "TFBS_ablation",
                                     "protein_altering_variant"])
-
-    def get_variant_annotations(self, id_list):
-        split = [id.split(':') for id in id_list]
-        variants = [{'chrom': s[0] if s[0].startswith('chr') else 'chr' + s[0], 'pos': int(s[1])} for s in split]
+        
+        datatypesf = Path(self.matrix_path + ".datatypes")
+        
+        def f(x):
+            return x
+
+        self.dconv = defaultdict( lambda: f )
+
+
+        print("Checking if type config exists {}".format( datatypesf ))
+        if ( datatypesf.is_file()):
+            print("Annotation datatype configuration file found. Using datatypes from {}".format(datatypesf) )
+            with datatypesf.open() as dtf:
+                # skip header.
+                l=dtf.readline()
+                for l in dtf:
+                    df = l.rstrip("\n").split()
+                    col = df[0]
+                    dtype = df[1]
+                    if dtype.lower() not in TabixAnnotationDao.DATA_CONVS:
+                        raise ConfigurationException("Unkown datatype given in datatype configuration file. Type given {}. Accepted types: {}".format(dtype, ",".join(list(TabixAnnotationDao.DATA_CONVS.keys()))) )
+                    self.dconv[col]=TabixAnnotationDao.DATA_CONVS[dtype]
+
+    def get_single_variant_annotations(self, variant:Variant) -> Variant:
+        res = self.get_variant_annotations([variant])
+        for r in res:
+            if r==variant:
+                return r
+        return None
+
+    def get_variant_annotations(self, variants:List[Variant]):
```

Comment:
Return the annotations and do not store them in variant annotations. That is callers job.